cvat-ui/src/components/login-form.tsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { FormComponentProps } from 'antd/lib/form/Form';
+import {
+    Button,
+    Icon,
+    Input,
+    Form,
+} from 'antd';
+
+export interface LoginData {
+    username: string;
+    password: string;
+}
+
+type LoginFormProps = {
+    onSubmit(loginData: LoginData): void;
+} & FormComponentProps;
+
+class LoginForm extends React.PureComponent<LoginFormProps> {
+    constructor(props: LoginFormProps) {
+        super(props);
+    }
+
+    private handleSubmit(e: React.FormEvent) {
```

Comment:
I am not really sure it's a good idea to use such a syntax. Because it isn't a part of specification for now, as far as I know.

Reply:
Don't worry it's a good practice in React community and is part of the ES7 specification you can enable it simply add @babel/plugin-proposal-class-properties.

And in any case, I don't recommend you use `bind` in render method if it not necessary, in the future, it will bad impact on performance, take a look at official documentation https://reactjs.org/docs/handling-events.html.